std/array.d

```diff
@@ -3889,3 +3889,50 @@ unittest
     assert(appS.data == "hellow");
     assert(appA.data == "hellow");
 }
+
+struct OrderedAA(K,V)
+{
+    struct KeyValue
+    {
+        K key;
+        V value;
+    }
+    private KeyValue[] array;
+    auto keyValues() inout { return array; }
+    alias byKeyValue = keyValues;
+
+    size_t length() const { return array.length; }
+    inout(V)* opIn(in K key) inout
+    {
+        foreach (ref member; array)
+            if (key == member.key)
+                return &member.value;
+        return null;
+    }
+    ref auto opIndex(in K key) const
+    {
+        foreach (ref member; array)
+        if (key == member.key)
+            return member.value;
+        assert(0, "OrderedAA does not contain value with key: " ~ key);
```

Comment:
* `~ key` will fail with non-string types
* unittest to catch that eg OrderedAA!(int, double)
* not sure if assert is appropriate; it probably should be same type of error as corresponding error in an  associative array